### ISSUE TITLE
[QUESTION] Parameter to disable caching while browsing?

### ISSUE THREAD

**Issue by codeOfJannik** [JaffaKetchup/flutter_map_tile_caching#40]
**What is the main problem?**
Is there a way to limit the costs of the tile server download by disabling caching while browsing? Because with caching while browsing enabled every tile that's displayed will be downloaded. And the higher the amount of downloaded tiles the higher the costs.

**Give more detail about what you need help with or what your question is**
I didn't find any flag or parameter to disable the caching while browsing. Is there anything implemented?

**Is there a potential for this question to turn into a feature request?**
If it's not implemented is that even possible? To add e.g. a bool _enableBrowseCaching_ that defaults to true, but if it's set to false and the device is online, tiles that have not been pre-downloaded will be loaded from the internet but not be cached.

**Comment by JaffaKetchup** [comment]
There is a way to disable caching, but this library doesn't provide it.
The way I do it in the example is to just use the ternary operator (?:) to choose between `StorageCachingTileProvider` and `NetworkTileProvider` (or whatever it is that you want), basing it on a variable you manage the state of.
This is because it would just make this library unnecessarily bigger.

If this doesn't work for you, I'll see what I can do.

At the same time though, browse caching shouldn't cost more than non browse caching. The tile is only downloaded once, not twice. The same downloaded bytes displayed on the map are also cached. If you have evidence of your tile server charging you extra, please attach a screenshot.

Thanks for your interest, hope this answers your question!

**Comment by codeOfJannik** [comment]
Okay, thank you for your reply and suggestions. I'll ask my colleague again who's in charge for our Mapbox account. He notified me a few days ago about the high charges.

**Comment by JaffaKetchup** [comment]
Ok no problem. If possible, send a before and after screenshot(s).
Meanwhile, I'll double check the network calls.

**Comment by JaffaKetchup** [comment]
Hi there @codeOfJannik,
Have you had any news yet? Interested to hear!
Thanks 👍

**Comment by JaffaKetchup** [comment]
**Here's what I found:**
TLDR: There shouldn't be any difference in API costs by using this library. Logically, it shouldn't, and the evidence below seems to prove this as far as I can understand.

---

The first image is with caching enabled:

![WithCaching](https://user-images.githubusercontent.com/58115698/152609722-0f3bf6d2-00c4-4bfc-bc53-e0a032e9d690.jpg)

It's clear that there are twice the number of network requests, BUT I don't think the `InternetAddress()` calls are actually to the tile server or anything that will increase your costs - they have 101 (Switching Protocols) instead of 200 (OK), and they are not HTTP requests.

---

The second image is without caching (just changing the provider as I mentioned above):

![WithoutCaching](https://user-images.githubusercontent.com/58115698/152609724-35806771-9c14-4208-97be-ba3d45644d77.jpg)

---

Thank you for using this library, and I hope it isn't causing you too many issues at the moment - there have been some performance issues, as well as potentially this.

**Comment by codeOfJannik** [comment]
I talked to my colleague and he checked the Mapbox pricing again. According to him, online used tiles don't cost apparently and tiles downloaded in bulk downloading clearly do cost. But he was not able to figure out if online tiles that are cached during browsing are also billed by Mapbox. The Mapbox pricing section is somehow confusing...
So not really news, I'm sorry.

**Comment by codeOfJannik** [comment]
But in addition to your post. Yes, we also assumed the same behavior, what you discovered, while we watched the costs correlated to the app use.

**Comment by JaffaKetchup** [comment]
Ok. If it's not a security problem, can you (or your colleague) visit https://account.mapbox.com/statistics, and send a screenshot of the APIs section, such as below?
![image](https://user-images.githubusercontent.com/58115698/152611073-961c1ab5-4bdd-4a9e-a662-79f71f9238c1.png)

Potentially, the bulk downloading is pushing you over some sort of margin? Note that setting up Mapbox with 'flutter_map' will result in costs to Static Tiles API, which are considerably more expensive than other options. This library also uses that API.

I can try to help you figure this out if you want - always happy to help! - but I can't take responsibility.

> he was not able to figure out if online tiles that are cached during browsing are also billed by Mapbox

I can confirm that browse caching tiles will make one request to Mapbox per tile - as usual. It won't change costs. The only thing that might substantially change costs is bulk downloading, if it pushes you over a margin. I'm not sure whether Mapbox charges for heavy sudden usage.

> The Mapbox pricing section is somehow confusing...

Agreed!

**Comment by JaffaKetchup** [comment]
Ok, I'm going to close this for now. Please feel free to add more information and ask for help if you need!
